homeassistant/components/goalzero/sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Support for Goal Zero Yeti."""
+import logging
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from requests import get, exceptions
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_HOST,
+    CONF_MONITORED_VARIABLES,
+    CONF_BINARY_SENSORS,
+    POWER_WATT,
+    ENERGY_WATT_HOUR,
+    ATTR_VOLTAGE,
+    TEMP_CELSIUS,
+    TIME_MINUTES,
+    TIME_SECONDS,
+    UNIT_PERCENTAGE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+_THROTTLED_REFRESH = None
+
+PLATFORMS = ["binary_sensor", "sensor"]
+
+SENSOR_TYPES = {
+    "thingName": ["Name", CONF_NAME],
+    "backlight": ["Backlight", CONF_BINARY_SENSORS],
+    "app_online": ["App Online", CONF_BINARY_SENSORS],
+    "wattsIn": ["Watts In", POWER_WATT],
+    "ampsIn": ["Amps In", "A"],
+    "wattsOut": ["Watts Out", POWER_WATT],
+    "ampsOut": ["Amps Out", "A"],
+    "whOut": ["Wh Out", ENERGY_WATT_HOUR],
+    "whStored": ["Wh Stored", ENERGY_WATT_HOUR],
+    "volts": ["Volts", ATTR_VOLTAGE],
+    "socPercent": ["State of Charge Percent", UNIT_PERCENTAGE],
+    "isCharging": ["Is Charging", CONF_BINARY_SENSORS],
+    "timeToEmptyFull": ["Time to Empty/Full", TIME_MINUTES],
+    "temperature": ["temperature", TEMP_CELSIUS],
+    "wifiStrength": ["Wifi Strength", "dB"],
+    "timestamp": ["Time Stamp", TIME_SECONDS],
+    "firmwareVersion": ["Firmware Version", None],
+    "version": ["Model Version", None],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.matches_regex(
+            r"\A(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2 \
+            [0-4][0-9]|[01]?[0-9][0-9]?)\Z"
+        ),
+        vol.Optional(CONF_NAME, default="Yeti"): cv.string,
+        vol.Optional(CONF_MONITORED_VARIABLES, default=[]): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES)]
+        ),
+    }
+)
```

Comment:
We no longer accept YAML configuration for integrations that connect to devices or services.

Instead, a configuration flow via the UI should be used.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

To get started on the integration configuration, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Reply:
That is in the works. This is my first contribution so any help is appreciated.

Reply:
If you need help or have questions, the best place to go would be our #dev_* channels on our Discord chat server.